src/main/java/com/tictactoe/Grid.java

```diff
@@ -0,0 +1,121 @@
+        ArrayList<Integer> symbolLength = new ArrayList<>();
+        for (int i = 0; i < boardArray.length; i++) {
+            symbolLength.add(boardArray[i].length());
+        }
+        return Collections.max(symbolLength);
+    }
+
+    private String getPaddedSymbols(String symbol, Integer maxLength) {
+        if (symbol.length() == maxLength) {
+            return symbol;
+        } else {
+            int lengthDifference = maxLength - symbol.length();
+            return this.padLeft(symbol, lengthDifference);
+        }
+    }
+
+    private String padLeft(String symbol, int space) {
+        space = space + 1;
+        return String.format("%1$" + space + "s", symbol);
+    }
+
+    public String getGrid() {
+        StringBuilder grid = new StringBuilder ("");
+        for (int cellIndex = 0; cellIndex <= this.boardSize - this.cellsPerRow; cellIndex += this.cellsPerRow){
+            grid.append(drawARow(cellIndex));
+            if (cellIndex < this.boardSize - this.cellsPerRow) {
+                grid.append(drawRowDivider());
+            }
+        }
+        grid.append("\n");
+        return grid.toString();
+    }
+
+    private String drawARow(int startingCell) {
+        StringBuilder row = new StringBuilder ("");
+        int maxLengthOfSymbols = this.findMaxStringLengthOfSymbols();
+        for (int cellIndex = startingCell; cellIndex < startingCell + cellsPerRow; cellIndex++) {
+              String lastCellCharacter = (cellIndex % cellsPerRow == this.cellsPerRow - 1)? " " : " |";
+              row.append(" " + this.getPaddedSymbols(board.getSymbol(cellIndex), maxLengthOfSymbols) + lastCellCharacter);
+        }
+        return row.toString();
+    }
+
+    private String drawRowDivider() {
+        String horizontalLine = "\n";
+        for (int i = 0; i < this.cellsPerRow; i++) {
+            horizontalLine += this.drawingDividerUnit();
+            horizontalLine += " ";
+        }
+        horizontalLine += "\n";
+        return horizontalLine;
+    }
+
+    private String drawingDividerUnit() {
+        String unit = "";
+        int maxLengthOfSymbols = this.findMaxStringLengthOfSymbols();
+        for (int j = 0; j < maxLengthOfSymbols + 2; j++) {
+            unit += "-";
+        }
+        return unit;
+    }
+
+    private String drawSymbolPadding() {
+        String unit = "";
+        int maxLengthOfSymbols = this.findMaxStringLengthOfSymbols();
+        for (int j = 0; j < maxLengthOfSymbols; j++) {
+            unit += " ";
+        }
+        return unit;
+    }
+
+    public void drawGrid() {
+        System.out.println(getGrid());
+    }
+
+
+    private String drawARowWithCellId(int startingCell) {
+        StringBuilder row = new StringBuilder ("");
+        for (int i = startingCell; i < startingCell + cellsPerRow; i++) {
+            if (i % cellsPerRow == this.cellsPerRow - 1) {
+                row.append(this.drawSymbolPadding() + (i+1) + " ");
```

Comment:
This is a minor nitpick. I notice that you have +1 and -1 scattered throughout your code. Off-by-one errors are very difficult to find and fix when your codebase gets bigger. I would recommend creating a constant called 'CELL_OFFSET` and then say `i + CELL_OFFSET` or `this.cellsPerRow - CELL_OFFSET`. This will get rid of the magic number